packages/admin/admin/src/form/FinalFormInput.tsx

```diff
@@ -1,15 +1,20 @@
-import { Translate } from "@comet/admin-icons";
-import { Button, InputBase, InputBaseProps, Tooltip } from "@mui/material";
+import { ArtificialIntelligence, Translate } from "@comet/admin-icons";
+import { Button, IconButton, InputBase, InputBaseProps, Tooltip } from "@mui/material";
+import { useTheme } from "@mui/material/styles";
 import * as React from "react";
-import { FieldRenderProps } from "react-final-form";
+import { FieldInputProps, FieldRenderProps } from "react-final-form";
 import { FormattedMessage } from "react-intl";
 
 import { ClearInputAdornment } from "../common/ClearInputAdornment";
+import { Loading } from "../common/Loading";
+import { useMlService } from "../ml/useMlService";
 import { useContentTranslationService } from "../translator/useContentTranslationService";
 
 export type FinalFormInputProps = InputBaseProps &
     FieldRenderProps<string, HTMLInputElement | HTMLTextAreaElement> & {
         clearable?: boolean;
+        mlButton?: (input: FieldInputProps<string, HTMLInputElement | HTMLTextAreaElement>) => void;
+        loading?: boolean;
```

Comment:
I don't like this. We should come up with a better solution.

Reply:
yes, there should be nothing "ml" specific in our `FinalFormInput. the possibility for an endAdornment should be enough so you can add the button and everything.

Reply:
I did this to expose the `input` and especially the `input.onChange` function. I need this function to change the content of the `textField` after generating new text. If I don't parse the input, I can use a `FormSpy` to achieve the same effect, but this is way more code and more complex. 
I think I could combine the `loading` into my `contentGenerationButton` by parsing a `Promise`. This would reduce it to just one prop. I could also parse the `input` at the `endAdornment`, then I would not need the `contentGenerationButton` prop.

Reply:
@Flips2001 now solved it using the Form Api and endAdornment. I think that's a good solution.